models/serialisation.ts

```diff
@@ -80,411 +82,446 @@ export const serializeInstructionToBase64 = (
-        ['config', GovernanceConfig],
-      ],
-    },
-  ],
-  [
-    CreateProposalArgs,
-    {
-      kind: 'struct',
-      fields: [
-        ['instruction', 'u8'],
-        ['name', 'string'],
-        ['descriptionLink', 'string'],
-        ['governingTokenMint', 'pubkey'],
-      ],
-    },
-  ],
-  [
-    AddSignatoryArgs,
-    {
-      kind: 'struct',
-      fields: [
-        ['instruction', 'u8'],
-        ['signatory', 'pubkey'],
-      ],
-    },
-  ],
-  [
-    SignOffProposalArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    CancelProposalArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    RelinquishVoteArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    FinalizeVoteArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    CastVoteArgs,
-    {
-      kind: 'struct',
-      fields: [
-        ['instruction', 'u8'],
-        ['vote', 'u8'],
-      ],
-    },
-  ],
-  [
-    InsertInstructionArgs,
-    {
-      kind: 'struct',
-      fields: [
-        ['instruction', 'u8'],
-        ['index', 'u16'],
-        ['holdUpTime', 'u32'],
-        ['instructionData', InstructionData],
-      ],
-    },
-  ],
-  [
-    RemoveInstructionArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    ExecuteInstructionArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    FlagInstructionErrorArgs,
-    {
-      kind: 'struct',
-      fields: [['instruction', 'u8']],
-    },
-  ],
-  [
-    SetRealmAuthorityArgs,
-    {
-      kind: 'struct',
-      fields: [
-        ['instruction', 'u8'],
-        ['newRealmAuthority', { kind: 'option', type: 'pubkey' }],
-      ],
-    },
-  ],
-  [
-    SetRealmConfigArgs,
-    {
-      kind: 'struct',
-      fields: [
-        ['instruction', 'u8'],
-        ['configArgs', RealmConfigArgs],
-      ],
-    },
-  ],
-  [
-    InstructionData,
-    {
-      kind: 'struct',
-      fields: [
-        ['programId', 'pubkey'],
-        ['accounts', [AccountMetaData]],
-        ['data', ['u8']],
-      ],
-    },
-  ],
-  [
-    AccountMetaData,
-    {
-      kind: 'struct',
-      fields: [
-        ['pubkey', 'pubkey'],
-        ['isSigner', 'u8'],
-        ['isWritable', 'u8'],
-      ],
-    },
-  ],
+export const GOVERNANCE_SCHEMA = createGovernanceSchema(ProgramVersion.V1)
+export const GOVERNANCE_SCHEMA_V2 = createGovernanceSchema(ProgramVersion.V2)
 
-  [
-    MintMaxVoteWeightSource,
-    {
-      kind: 'struct',
-      fields: [
-        ['type', 'u8'],
-        ['value', 'u64'],
-      ],
-    },
-  ],
+export function getGovernanceSchema(programVersion: ProgramVersion) {
+  switch (programVersion) {
+    case ProgramVersion.V1:
+      return GOVERNANCE_SCHEMA
+    default:
+      return GOVERNANCE_SCHEMA_V2
+  }
+}
```

Comment:
had to port this over from realm-explorer to have it work properly for V2 governance programs

Reply:
:+1: